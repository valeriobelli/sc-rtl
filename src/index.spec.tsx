import { render } from '@testing-library/react'
import type { PropsWithChildren } from 'react'
import styled, { ThemeProvider } from 'styled-components'
import { describe, expect, it } from 'vitest'
import { rtl } from '.'

describe('rtl', () => {
	it('lefts intact styles when in ltr', () => {
		const Component = styled.div`
			margin-left: 16px;
		`

		const Wrapper = ({ children }: PropsWithChildren) => (
			<ThemeProvider theme={{ dir: 'ltr' }}>{children}</ThemeProvider>
		)

		const { container } = render(<Component />, { wrapper: Wrapper })

		expect(container.firstChild).toMatchSnapshot()
	})

	it('lefts intact styles when in rtl not applied to "rtl" template literal', () => {
		const Component = styled.div`
			margin-left: 16px;
		`

		const Wrapper = ({ children }: PropsWithChildren) => (
			<ThemeProvider theme={{ dir: 'rtl' }}>{children}</ThemeProvider>
		)

		const { container } = render(<Component />, { wrapper: Wrapper })

		expect(container.firstChild).toMatchSnapshot()
	})

	it('reverses styles when in rtl and the styles are applied to "rtl" template literal', () => {
		const Component = styled.div<{ pl: number }>`
			${rtl`
				${({ theme: { dir } }) => (dir === 'ltr' ? 'text-align: left;' : 'text-align: right;')}

				margin-left: 16px;
				padding-right: 8px;
				padding-left: ${({ pl }) => pl}px;
				@keyframes foo {}
			`}
		`

		const Wrapper = ({ children }: PropsWithChildren) => (
			<ThemeProvider theme={{ dir: 'rtl' }}>{children}</ThemeProvider>
		)

		const { container } = render(<Component pl={4} />, { wrapper: Wrapper })

		expect(container.firstChild).toMatchSnapshot()
	})
})
