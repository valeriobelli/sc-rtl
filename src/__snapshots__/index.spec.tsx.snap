// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rtl > lefts intact styles when in ltr 1`] = `
.c0 {
  margin-left: 16px;
}

<div
  class="c0"
/>
`;

exports[`rtl > lefts intact styles when in rtl not applied to "rtl" template literal 1`] = `
.c0 {
  margin-left: 16px;
}

<div
  class="c0"
/>
`;

exports[`rtl > reverses styles when in rtl and the styles are applied to "rtl" template literal 1`] = `
.c0 {
  text-align: left;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 4px;
}

<div
  class="c0"
/>
`;
